From f20856e12e6ea7721607296c4e006b50332add56 Mon Sep 17 00:00:00 2001
From: Alaa Hleihel <ahleihel@redhat.com>
Date: Sun, 10 May 2020 15:04:43 -0400
Subject: [PATCH 118/312] [netdrv] net/mlx5e: Always print health reporter
 message to dmesg

Message-id: <20200510150452.10307-79-ahleihel@redhat.com>
Patchwork-id: 306702
Patchwork-instance: patchwork
O-Subject: [RHEL8.3 BZ 1789380 v2 78/87] net/mlx5e: Always print health reporter message to dmesg
Bugzilla: 1789380
RH-Acked-by: Kamal Heib <kheib@redhat.com>
RH-Acked-by: Jarod Wilson <jarod@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Jonathan Toppins <jtoppins@redhat.com>

Bugzilla: http://bugzilla.redhat.com/1789380
Upstream: v5.5-rc6

commit 99cda45426c9a2c59bb2f7cb886a405440282455
Author: Eran Ben Elisha <eranbe@mellanox.com>
Date:   Wed Dec 4 14:34:18 2019 +0200

    net/mlx5e: Always print health reporter message to dmesg

    In case a reporter exists, error message is logged only to the devlink
    tracer. The devlink tracer is a visibility utility only, which user can
    choose not to monitor.
    After cited patch, 3rd party monitoring tools that tracks these error
    message will no longer find them in dmesg, causing a regression.

    With this patch, error messages are also logged into the dmesg.

    Fixes: c50de4af1d63 ("net/mlx5e: Generalize tx reporter's functionality")
    Signed-off-by: Eran Ben Elisha <eranbe@mellanox.com>
    Signed-off-by: Saeed Mahameed <saeedm@mellanox.com>

Signed-off-by: Alaa Hleihel <ahleihel@redhat.com>
Signed-off-by: Frantisek Hrbata <fhrbata@redhat.com>
---
 drivers/net/ethernet/mellanox/mlx5/core/en/health.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/mellanox/mlx5/core/en/health.c b/drivers/net/ethernet/mellanox/mlx5/core/en/health.c
index 1d6b58860da6..3a975641f902 100644
--- a/drivers/net/ethernet/mellanox/mlx5/core/en/health.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/en/health.c
@@ -197,9 +197,10 @@ int mlx5e_health_report(struct mlx5e_priv *priv,
 			struct devlink_health_reporter *reporter, char *err_str,
 			struct mlx5e_err_ctx *err_ctx)
 {
-	if (!reporter) {
-		netdev_err(priv->netdev, err_str);
+	netdev_err(priv->netdev, err_str);
+
+	if (!reporter)
 		return err_ctx->recover(&err_ctx->ctx);
-	}
+
 	return devlink_health_report(reporter, err_str, err_ctx);
 }
-- 
2.13.6

